batcher/aligned-batcher/src/zk_utils/mod.rs

```diff
@@ -44,13 +46,17 @@ pub(crate) fn verify(verification_data: &VerificationData) -> bool {
             is_valid
         }
         ProvingSystemId::Risc0 => {
-            if let Some(image_id_slice) = &verification_data.vm_program_code {
-                let mut image_id = [0u8; 32];
-                image_id.copy_from_slice(image_id_slice.as_slice());
-                return verify_risc_zero_proof(verification_data.proof.as_slice(), &image_id);
+            if verification_data.vm_program_code.is_none() || verification_data.pub_input.is_none() {
+                warn!("Trying to verify Risc0 proof but image id or public input was not provided. Returning false");
+                return false;
             }
-            warn!("Trying to verify Risc0 proof but image ID was not provided. Returning false");
-            false
+
+            let image_id_slice = verification_data.vm_program_code.as_ref().unwrap().as_slice();
+            let public_input = verification_data.pub_input.as_ref().unwrap().as_slice();
+
+            let mut image_id = [0u8; 32];
+            image_id.copy_from_slice(image_id_slice);
+            return verify_risc_zero_proof(verification_data.proof.as_slice(), &image_id, public_input);
```

Comment:
small nit but would be ideal if we could catch the vm_program_code and pub_input inside a if let statement and avoid the unwraps

Reply:
I thought checking for none on top and having unwraps below was a bit cleaner than nested ifs, rust won't allow you to have if let _ = .. && let __ = ...

Reply:
changed on 5deeb9fd7f90f2f0dbe2a16db2551cbd375061ce